src/parser/mod.rs

```diff
@@ -33,21 +33,38 @@ impl Parser {
     /// [`proc-macro2::TokenStream`]: https://docs.rs/proc-macro2/latest/proc_macro2/struct.TokenStream.html
     /// [`proc-macro::TokenStream`]: https://doc.rust-lang.org/proc_macro/struct.TokenStream.html
     /// [`Node`]: struct.Node.html
-    pub fn parse_simple(&self, v: impl Into<TokenStream>) -> Result<Vec<Node>> {
+    pub fn parse_simple(&self, v: impl Into<TokenStream>) -> Result<Vec<Node<Infallible>>> {
+        self.parse_recoverable(v).into_result()
+    }
+
+    /// Parse the given [`proc-macro2::TokenStream`] or
+    /// [`proc-macro::TokenStream`] into a [`Node`] tree with the [`CustomNode`]
+    /// C.
+    ///
+    /// [`proc-macro2::TokenStream`]: https://docs.rs/proc-macro2/latest/proc_macro2/struct.TokenStream.html
+    /// [`proc-macro::TokenStream`]: https://doc.rust-lang.org/proc_macro/struct.TokenStream.html
+    /// [`Node`]: struct.Node.html
+    pub fn parse_custom<C: CustomNode>(&self, v: impl Into<TokenStream>) -> Result<Vec<Node<C>>> {
```

Comment:
I recommend moving <C> to `ParserConfig` (and `Parser`) so that interface methods would remain the same.

Optionally: We also need method in `ParserConfig` to specify custom node, something like this:
```rust
fn with_custom_node::<C: CustomNode>(mut self) -> ParserConfig<C> {
...
}
// We also can add adapter, so library user can avoid implementing `CustomNode` and use ``Parse`/`ToTokens` implementation instead.
`fn with_custom_node_syn_parser<T:Parse+ToTokens>(mut self) -> ParserConfig<Wrapper<T>>`
```

Reply:
nether mind my question (https://github.com/rs-tml/rstml/pull/34#discussion_r1350546909), I just noticed this :D

Reply:
True, though how would peeking happen? Just by running `parse`?

Reply:
> > ```rust
> > // We also can add adapter, so library user can avoid implementing `CustomNode` and use ``Parse`/`ToTokens` implementation instead.
> > `fn with_custom_node_syn_parser<T:Parse+ToTokens>(mut self) -> ParserConfig<Wrapper<T>>`
> > ```
> 
> True, though how would peeking happen? Just by running `parse`?

yes, its okay to use parse, if we use .fork() on the call side